### ISSUE TITLE
Feature request: Portrait Support.

### ISSUE THREAD

**Issue by LoganFraser** [terjeio/ioSender#277]
There's a bunch of low-cost portable touchscreens with 1080p (1920x1080) that would be great mounted to a CNC in portrait mode.  It's not wide enough for XL when I loaded it so using Standard, there's plenty of room to have the Jogger displayed in the middle at the top of the layout, console below that, and then the remaining half for either 3d or program view.

**Comment by terjeio** [comment]
You can try that?

The UI layout for the center part of the _Grbl_ tab [is here](https://github.com/terjeio/ioSender/blob/6a7f84b394e0469ccd15ea207d5024cd251f9df5/ioSender%20XL/ioSender%20XL/JobView.xaml#L68-L81).  Unwrap and remove the tab control and replace the dock panel with a vertical stack panel should work. You may keep the tab control without the console if you want both the 3D view and the program view. The jogger [is added here](https://github.com/terjeio/ioSender/blob/6a7f84b394e0469ccd15ea207d5024cd251f9df5/ioSender%20XL/ioSender%20XL/JobView.xaml#L56), and can be moved to the new stack panel.

**Comment by LoganFraser** [comment]
I appreciate your pointing out the locations. I'll see if I can get someone from the PrintNC discord to follow up.  I'm an electrician/hardware design guy, my programming skills are extremely limited.  Thanks for the reply.  :)

**Comment by mintracer** [comment]
Hi Terje,

so this is my first attempt in changing an UI in Visual Studio... My first touch with this program :)
I changed the JobView.xaml to a Grid Layout - 2 Colums and 3 Rows 
I hope you like it and can give me some hints how to make it better :) 

btw. The Gcode is just a double bottle opener... :D

![WhatsApp Image 2023-03-19 at 19 01 38](https://user-images.githubusercontent.com/97637453/226279694-2533b3f2-8a15-45c3-a580-200fdcadbf37.jpeg)
![WhatsApp Image 2023-03-19 at 19 06 17](https://user-images.githubusercontent.com/97637453/226279697-e72b4c69-e002-413b-bc57-18417d009323.jpeg)

**Comment by terjeio** [comment]
Looks good - but it is wide enough for the probing tabs?

Instead of two columns at the top three will be better? At least with more than 3 axes configured? IMO _Program limits_ could be moved to the center column.

**Comment by mintracer** [comment]
todays progress... :)

![Screenshot 2023-03-20 212031](https://user-images.githubusercontent.com/97637453/226459304-6b79a95f-8116-4e01-b258-0cb62438d197.png)

**Comment by terjeio** [comment]
I have to admit that I am old fashioned and do not fancy such colorful displays. Users younger than me migh have a different opinion - and I am ok with that. Hopefully someone else can comment too.

**Comment by mmaczollek** [comment]
I would love a dark theme to the program. I spend a lot of time on computers for work and the light background gives me migraines. @mintracer, can you share that UI somehow? It looks really great.

**Comment by mintracer** [comment]
> I would love a dark theme to the program. I spend a lot of time on computers for work and the light background gives me migraines. @mintracer, can you share that UI somehow? It looks really great.

it´s a really early preview... since i´m not a professional coder as Terje is... I´m learning all the wpf theming... so it will take a while before any "sharing" will be possible.. and offcourse Terje has the last word on it... it is his "baby" ;) 

@terjeio is there a central place for all the coloring in wpf? such as css on websites?

**Comment by terjeio** [comment]
> is there a central place for all the coloring in wpf? such as css on websites?

I have not read up on [theming in WPF apps](https://duckduckgo.com/?t=ffab&q=wpf+theming&ia=web) so I do not know what is required nor how it is done

**Comment by axg20202** [comment]
Hi all. Sorry to revive this old thread, but is anyone able to share a portrait layout (source files, preferably), in default colours as per first image above?

**Comment by LoganFraser** [comment]
There's a touch focused fork of iosender that has more direct support for portrait.  My machine is currently in pieces so I haven't looked at it in detail yet but I'm impressed with what I have seen.  https://github.com/Jay-Tech/ioSenderTouch/tree/IOSenderTouch

**Comment by LoganFraser** [comment]
Didn't mean to close it.  ;)

**Comment by axg20202** [comment]
Thanks. The issue is that I'm not using a touchscreen. Willing to consider switching to one though if it really is a good move.

**Comment by phil-barrett** [comment]
I'm looking at using a touch screen.  A lot of people hate the idea of hooking up a PC to their CNC machine but love the idea of the Masso controller which is effectively the same with a touch screen.  I think that using a breakoutboard, microPC and touchscreen could be about 75% of the cost of a Masso and way more flexible.  Will be protoyping this fairly soon.

**Comment by axg20202** [comment]
Are there touchscreens that are recommended or those to avoid? I've seen some comments on avoiding capacitive screens, but I think those are the affordable ones.

**Comment by phil-barrett** [comment]
actually, it is resistive touch that is slow an imprecise.  cap is better.  

I am looking for a windows touch screen that works with a "stock" miniPC.  I believe they work by using a USB connection to emulate mouse/keyboard.

**Comment by axg20202** [comment]
OK, good to know. I hope a capacitive screen won't go nuts as soon as my vfd fires up. I've seen quite a few all in one pc that have the guts of a mini pc built into the touchscreen too. I.e. Hp G2 touchscreen, Dell made some too.

**Comment by phil-barrett** [comment]
Yes, the all-in-one PCs will probably work ok but they tend to be expensive. I am seeing miniPCs that are in the $200-300 USD range that should work pretty well.

**Comment by LoganFraser** [comment]
I'm just using one of the portable USB-C powered 15" touchscreens available on Amazon.  It works well enough.

**Comment by axg20202** [comment]
I've decided against touch screen. Can anyone help me out with editing the job view xaml of the standard size iosender? I really like the first portrait layout shown towards the top of this thread. Even that job view file would be a huge help for me to get started on tweaking it further.

**Comment by axg20202** [comment]
Just bumping this thread. Terje, is this something that you could support please? Being able to view ioSender in portrait would be hugely helpful but I am completely at sea when it comes to editing the code to achieve this.

**Comment by andrewmarles** [comment]
Hopefully @terjeio it is ok to post, but a user has done a beautiful port of the iosender codebase that is optimized for touch and has portrait support.

https://github.com/Jay-Tech/ioSenderTouch
